src/agent/src/network.rs

```diff
@@ -28,3 +37,116 @@ impl Network {
         self.dns.push(dns);
     }
 }
+
+pub fn setup_guest_dns(logger: Logger, dns_list: Vec<String>) -> Result<()> {
+    do_setup_guest_dns(
+        logger,
+        dns_list,
+        KATA_GUEST_SANDBOX_DNS_FILE,
+        GUEST_DNS_FILE,
+    )
+}
+
+fn do_setup_guest_dns(logger: Logger, dns_list: Vec<String>, src: &str, dst: &str) -> Result<()> {
+    let logger = logger.new(o!( "subsystem" => "network"));
+
+    if dns_list.len() == 0 {
+        info!(
+            logger,
+            "Did not set sandbox DNS as DNS not received as part of grpc request."
+        );
+        return Ok(());
+    }
+
+    let attr = fs::metadata(dst);
+    if attr.is_err() {
+        // not exists or other errors that we could not use it anymore.
+        return Ok(());
+    }
+
+    if attr.unwrap().is_dir() {
```

Comment:
I think it would be safer to check that the file is the expected type (normal file? symlinks allowed?) and error otherwise.

Reply:
Through https://doc.rust-lang.org/std/fs/struct.FileType.html I think filter out directory may be safe. Regular files and symlinks  are OK to mount.